commercial/app/model/commercial/CapiAgent.scala

```diff
@@ -0,0 +1,47 @@
+package model.commercial
+
+import common.AkkaAgent
+import model.Content
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+object CapiAgent {
+
+  private lazy val shortUrlAgent = AkkaAgent[Map[String, Option[Content]]](Map.empty)
+
+  def contentByShortUrls(shortUrls: Seq[String])
+                        (implicit ec: ExecutionContext): Future[Seq[Content]] = {
+
+    val shortUrlIds = shortUrls map { url =>
+      val slashPrefixed = if (url startsWith "/") url else s"/$url"
+      val endTrimmed = slashPrefixed.take(slashPrefixed.indexOf("/", 3))
+      endTrimmed
+    }
+
+    val cache = shortUrlAgent.get()
+
+    val urlsNotInCache = shortUrlIds filterNot cache.contains
+
+    def addToCache(contents: Seq[Content]): Future[Map[String, Option[Content]]] = {
```

Comment:
Why not have this inside the `object`? It's a matter of taste but I didn't spot it straight away that all these statements were inside a `def`!